aries_cloudagent/core/in_memory.py

```diff
@@ -49,6 +49,11 @@ def bind_providers(self):
             ),
         )
 
+    def _init_context(self):
```

Comment:
I don't see why this is necessary, it's already performed for the session when that is created?

Reply:
This is performed on the session only when `test_session ` function is called. But in this case we call `test_profile`.

Another option would be to call init_context every time we create a new session (in constructor) .

Reply:
The session must be awaited or used as an async context manager. When this happens the `_setup` method is called, which calls `_init_context` on the session. If the session has not been initialized this way then it is not technically 'open' (the in-memory backend is more forgiving about this currently). I think that adding this workaround is just masking the issue.

Reply:
Thanks for the explanation. We removed `_init_context` on the profile and created sessions with async context.